### DISCUSSION TITLE
End-of-life'd .NET support

### DISCUSSION THREAD

**Discussion by kfcampbell** [octokit/octokit.net#2748] [General]
Currently, we have several repositories that [still use .NET 5](https://github.com/octokit/octokit.net/blob/main/.github/workflows/publish.yml#L16-L19) and other previous end-of-life'd versions of .NET (see the .NET [support calendar here](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core)). 

I think it would be appropriate to make .NET 6 the floor of our versions and shoot to build and publish using .NET 6 everywhere, as it's the current LTS version and will be supported until November 2024. Thoughts?

**Comment by timheuer** [comment]
Depending on how long you anticipate this effort to take, .NET8 will ship in Nov this year...better to jump on that with maybe using v6 as a prepatory stepping stone to identify if at all any breaking changes to accommodate.

**Reply by kfcampbell** [reply]
Ooh, good catch! Waiting for .NET 8 seems like a reasonable approach to me.

**Reply by timheuer** [reply]
If there is an effort starting now, we just released Preview 7 for .NET 8 and can start making the move toward that to be immediately ready for RC/release ;-)

**Reply by kfcampbell** [reply]
Realistically, GitHub's SDK team will not get to this in the immediate/near future. I think it's likely that we can easily wait until November.

**Comment by eriawan** [comment]
@kfcampbell 

> Ooh, good catch! Waiting for .NET 8 seems like a reasonable approach to me.

I suggest we can start to support .NET 7 first now, because I think having .NET 7 support also means that we will be more prepared for .NET 8 after .NET 8 is released in November.
Let me know what do you think.

**Reply by kfcampbell** [reply]
I hear that, I just don't think realistically my team will be able to slot both of those upgrades in in time.

**Reply by eriawan** [reply]
Makes sense. Putting more focus on .NET 8 seems reasonable and since it is also LTS, it gives us enough time to keep supporting it.
